dlib: cannot use library from cmake

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
Run brew config
  
HOMEBREW_VERSION: 4.2.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: eb7338abba43112e16e178d4329d5483d6d8fd6f
Last commit: 3 weeks ago
Core tap JSON: 18 Feb 21:07 UTC
Core cask tap JSON: 18 Feb 21:07 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_COLOR: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 14.0.0 build 1400
Git: 2.43.2 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 12.7.3-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2 => /Applications/Xcode_14.2.app/Contents/Developer


Run brew doctor
  
Your system is ready to brew.



```



### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I'm trying to use dlib in a c++ project from cmake

### What happened (include all command output)?

The compilation fails using a basic optimization example from the dlib repo to isolate the problem:
```
Run cmake -B build -DCMAKE_VERBOSE_MAKEFILE=ON .
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found dlib: /usr/local/lib/cmake/dlib (found version 19.24.2)
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Found BLAS: /usr/local/opt/openblas/lib/libopenblas.dylib  
-- Looking for cheev_
-- Looking for cheev_ - found
-- Found LAPACK: /usr/local/opt/openblas/lib/libopenblas.dylib;-lm;-ldl  
-- Configuring done (3.6s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build
Change Dir: '/Users/runner/work/dlib-cmake-example/dlib-cmake-example/build'
Run Build Command(s): /usr/local/Cellar/cmake/3.28.3/bin/cmake -E env VERBOSE=1 /usr/bin/make -f Makefile
/usr/local/Cellar/cmake/3.28.3/bin/cmake -S/Users/runner/work/dlib-cmake-example/dlib-cmake-example -B/Users/runner/work/dlib-cmake-example/dlib-cmake-example/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.28.3/bin/cmake -E cmake_progress_start /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build/CMakeFiles /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build//CMakeFiles/progress.marks
/Applications/Xcode_14.2.app/Contents/Developer/usr/bin/make  -f CMakeFiles/Makefile2 all
/Applications/Xcode_14.2.app/Contents/Developer/usr/bin/make  -f CMakeFiles/optim_ex.dir/build.make CMakeFiles/optim_ex.dir/depend
cd /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build && /usr/local/Cellar/cmake/3.28.3/bin/cmake -E cmake_depends "Unix Makefiles" /Users/runner/work/dlib-cmake-example/dlib-cmake-example /Users/runner/work/dlib-cmake-example/dlib-cmake-example /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build /Users/runner/work/dlib-cmake-example/dlib-cmake-example/build/CMakeFiles/optim_ex.dir/DependInfo.cmake "--color="
/Applications/Xcode_14.2.app/Contents/Developer/usr/bin/make  -f CMakeFiles/optim_ex.dir/build.make CMakeFiles/optim_ex.dir/build
[ 25%] Building CXX object CMakeFiles/optim_ex.dir/optimization_ex.cpp.o
/Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -isystem /usr/local/include -isystem /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include -O3 -DNDEBUG -std=gnu++14 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -mmacosx-version-min=12.7 -MD -MT CMakeFiles/optim_ex.dir/optimization_ex.cpp.o -MF CMakeFiles/optim_ex.dir/optimization_ex.cpp.o.d -o CMakeFiles/optim_ex.dir/optimization_ex.cpp.o -c /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:6:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:642:26: error: no template named 'numeric_limits'
    bool _FloatBigger = (numeric_limits<_FloatT>::digits > numeric_limits<_IntT>::digits),
                         ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:642:60: error: no template named 'numeric_limits'
    bool _FloatBigger = (numeric_limits<_FloatT>::digits > numeric_limits<_IntT>::digits),
                                                           ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:643:18: error: no template named 'numeric_limits'
    int _Bits = (numeric_limits<_IntT>::digits - numeric_limits<_FloatT>::digits)>
                 ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:643:50: error: no template named 'numeric_limits'
    int _Bits = (numeric_limits<_IntT>::digits - numeric_limits<_FloatT>::digits)>
                                                 ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:648:17: error: no template named 'numeric_limits'
  static_assert(numeric_limits<_FloatT>::radix == 2, "FloatT has incorrect radix");
                ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:651:25: error: no template named 'numeric_limits'
  return _FloatBigger ? numeric_limits<_IntT>::max() :  (numeric_limits<_IntT>::max() >> _Bits << _Bits);
                        ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:651:58: error: no template named 'numeric_limits'
  return _FloatBigger ? numeric_limits<_IntT>::max() :  (numeric_limits<_IntT>::max() >> _Bits << _Bits);
                                                         ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:661:16: error: no template named 'numeric_limits'
  using _Lim = numeric_limits<_IntT>;
               ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:664:12: error: use of undeclared identifier '_Lim'
    return _Lim::max();
           ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:665:21: error: use of undeclared identifier '_Lim'
  } else if (__r <= _Lim::lowest()) {
                    ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:666:12: error: use of undeclared identifier '_Lim'
    return _Lim::min();
           ^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix_exp.h:6:
In file included from /usr/local/include/dlib/matrix/../algs.h:97:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/string:520:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__functional_base:22:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/exception:83:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cstdlib:138:9: error: target of using declaration conflicts with declaration already in scope
using ::abs _LIBCPP_USING_IF_EXISTS;
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/stdlib.h:132:6: note: target of using declaration
int      abs(int) __pure2;
         ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: conflicting declaration
using ::abs _LIBCPP_USING_IF_EXISTS;
^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix_exp.h:6:
/usr/local/include/dlib/matrix/../algs.h:493:21: error: expected unqualified-id
        return std::isfinite(value);
                    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/math.h:155:5: note: expanded from macro 'isfinite'
    ( sizeof(x) == sizeof(float)  ? __inline_isfinitef((float)(x))       \
    ^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix_exp.h:6:
/usr/local/include/dlib/matrix/../algs.h:499:21: error: expected unqualified-id
        return std::isfinite((double)value);
                    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/math.h:155:5: note: expanded from macro 'isfinite'
    ( sizeof(x) == sizeof(float)  ? __inline_isfinitef((float)(x))       \
    ^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:9:
In file included from /usr/local/include/dlib/matrix/../serialize.h:230:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:853:1: error: declaration conflicts with target of using declaration already in scope
abs(const complex<_Tp>& __c)
^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: target of using declaration
using ::abs _LIBCPP_USING_IF_EXISTS;
^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:9: note: using declaration
using ::abs _LIBCPP_USING_IF_EXISTS;
        ^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:9:
In file included from /usr/local/include/dlib/matrix/../serialize.h:230:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:910:16: error: reference to unresolved using declaration
        return abs(__c.real());
               ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: using declaration annotated with 'using_if_exists' here
using ::abs _LIBCPP_USING_IF_EXISTS;
^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:9:
In file included from /usr/local/include/dlib/matrix/../serialize.h:230:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:912:16: error: reference to unresolved using declaration
        return abs(__c.imag());
               ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: using declaration annotated with 'using_if_exists' here
using ::abs _LIBCPP_USING_IF_EXISTS;
^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:9:
In file included from /usr/local/include/dlib/matrix/../serialize.h:230:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:969:16: error: reference to unresolved using declaration
        __re = abs(__re);
               ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: using declaration annotated with 'using_if_exists' here
using ::abs _LIBCPP_USING_IF_EXISTS;
^
In file included from /Users/runner/work/dlib-cmake-example/dlib-cmake-example/optimization_ex.cpp:15:
In file included from /usr/local/include/dlib/optimization.h:6:
In file included from /usr/local/include/dlib/optimization/optimization.h:9:
In file included from /usr/local/include/dlib/optimization/optimization_search_strategies.h:8:
In file included from /usr/local/include/dlib/optimization/../matrix.h:6:
In file included from /usr/local/include/dlib/matrix/matrix.h:9:
In file included from /usr/local/include/dlib/matrix/../serialize.h:230:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:1022:29: error: reference to unresolved using declaration
    return complex<_Tp>(log(abs(__x)), arg(__x));
                            ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/cmath:338:1: note: using declaration annotated with 'using_if_exists' here
using ::abs _LIBCPP_USING_IF_EXISTS;
^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [CMakeFiles/optim_ex.dir/optimization_ex.cpp.o] Error 1
make[1]: *** [CMakeFiles/optim_ex.dir/all] Error 2
make: *** [all] Error 2
```

### What did you expect to happen?

compilation should succeed

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install dlib
git clone https://github.com/jschueller/dlib-cmake-example.git
cd dlib-cmake-example
cmake -B build .
cmake --build build
```

I created a repo with github actions enabled on osx to reproduce:
https://github.com/jschueller/dlib-cmake-example/actions/runs/8114102470/job/22178970286

Also If I remove 
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include from dlib configuration file (dlib.cmake) the compilation passes
maybe this path is outdated and dlib should be rebuilt or something ?

the job passes if I drop this path:
https://github.com/jschueller/dlib-cmake-example/blob/main/.github/workflows/build.yml#L40

--- a/Formula/d/dlib.rb
+++ b/Formula/d/dlib.rb
@@ -12,13 +12,14 @@ class Dlib < Formula
   end
 
   bottle do
-    sha256 cellar: :any,                 arm64_sonoma:   "22eab3f6edb254c1d4e2ded48f795bad876f130f03d81edbc4679ec66136820c"
-    sha256 cellar: :any,                 arm64_ventura:  "6d6cae37f9439f4bbaa534412c3ec957ed8da0c9a14febdc387eae7d8e9a96dd"
-    sha256 cellar: :any,                 arm64_monterey: "773966ac744aada4490deba607795cbbf4a506af2a3fa4196045f613fffcbb55"
-    sha256 cellar: :any,                 sonoma:         "3882169b190fd2f7eda3d39c4f1fd9800f2e610ee5e058a91e0613574eea85db"
-    sha256 cellar: :any,                 ventura:        "f0a79ecbfd60fba3650eb41c2abe5d93a3dae2a1756102bcc0b524cab2dbd91c"
-    sha256 cellar: :any,                 monterey:       "8c67247601b0b4d2824e0eaac979c7f22d1ac43ab3b1cb55d765775b1e90d27d"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "548d76fa95dc4639132aef2f2637db129f6c0a94d8e15f57db34b8fe4c580345"
+    rebuild 1
+    sha256 cellar: :any,                 arm64_sonoma:   "f382c1f7935c4bcd5000bf77e4b96571798713e7adc45c131b5cd53386e6e6d0"
+    sha256 cellar: :any,                 arm64_ventura:  "09dac41cac342b4057f1e0422fe810d6eb92cd07970ac97c221b17fe836c12a7"
+    sha256 cellar: :any,                 arm64_monterey: "426349c468344255a1f03a63a5d707b83f8837c44e35b46920cca3b7fbaaa2e5"
+    sha256 cellar: :any,                 sonoma:         "57059294f7ca3a47e5055ab12b0cbc67ca842832f88609ec8d2d375bcedf73f3"
+    sha256 cellar: :any,                 ventura:        "17e51002247200dcd930d7d3677e92eb71538aac1f4afc446ac3a12bd7f680a3"
+    sha256 cellar: :any,                 monterey:       "0f6ad77f551f870b05c6669e3b987f35dec7c4cd51564697e771d146d37bb7ea"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "14de2a87642ed886204d1548944a726b72f6df827199996f22f8548b1afa122e"
   end
 
   depends_on "cmake" => :build

--- a/Formula/d/dlib.rb
+++ b/Formula/d/dlib.rb
@@ -33,6 +33,7 @@ class Dlib < Formula
       -Dcblas_lib=#{Formula["openblas"].opt_lib/shared_library("libopenblas")}
       -Dlapack_lib=#{Formula["openblas"].opt_lib/shared_library("libopenblas")}
       -DDLIB_NO_GUI_SUPPORT=ON
+      -DDLIB_LINK_WITH_SQLITE3=OFF
       -DBUILD_SHARED_LIBS=ON
     ]
 

